miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/SampleRestController.java

```diff
@@ -203,6 +203,32 @@ public DataTablesResponseDto<SampleDto> getDTArrayedSamplesByProject(@PathVariab
         PaginationFilter.arrayed(true));
   }
 
+  @GetMapping(value = "/dt/samples-prepared/{id}", produces = {"application/json"})
+  @ResponseBody
+  public DataTablesResponseDto<SampleDto> getPreparedSamplessByRequisition(@PathVariable("id") long id,
+      HttpServletRequest request) throws IOException {
+    log.debug("Fetching derivatives for requisition: {}", id);
+
+    List<Sample> requisitionedSamples = sampleService.list(0, 0, false, "id", PaginationFilter.requisitionId(id));
+    List<Sample> supplementalSamples =
+        sampleService.list(0, 0, false, "id", PaginationFilter.supplementalToRequisitionId(id));
+
+    Set<Long> sampleIds = Stream.concat(requisitionedSamples.stream(), supplementalSamples.stream())
+        .map(Sample::getId)
+        .collect(Collectors.toSet());
+
+    List<Sample> stockSamples = sampleService.getChildren(sampleIds, SampleStock.CATEGORY_NAME, id);
+    List<Sample> tissueSamples = sampleService.getChildren(sampleIds, SampleTissue.CATEGORY_NAME, id);
+    List<Sample> aliquotSamples = sampleService.getChildren(sampleIds, SampleAliquot.CATEGORY_NAME, id);
```

Comment:
Missing tissue processing. Can also improve performance by going top-down (tissue > processing > stock > aliquot), mapping to IDs immediately since that's all you need, and searching based on the previous level. e.g. to find aliquots, you can use the stock IDs so it doesn't have to search through all the upper levels of the hierarchy to get there.